internal/builtinfunctions/functions_test.go

```diff
@@ -819,3 +822,194 @@ func Test_floatToFormattedString_success(t *testing.T) {
 		}
 	}
 }
+
+func expectedOutBindConstants(repInp any, items []any) []any {
+	return []any{
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[0], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[1], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[2], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+	}
+}
+
+func Test_bindConstants(t *testing.T) {
+	items := []any{
+		map[string]any{"loop_id": 1},
+		map[string]any{"loop_id": 2},
+		map[string]any{"loop_id": 3},
+	}
+	repeatedInputsMap := map[string]any{
+		"a": "A", "b": "B",
+	}
+	repeatedInputsInt := 2
+	repeatedInputsRegexPattern := regexp.MustCompile("p([a-z]+)ch")
+	testItems := map[string]any{
+		"combine with map":   repeatedInputsMap,
+		"combine with list":  items,
+		"combine with int":   repeatedInputsInt,
+		"combine with regex": repeatedInputsRegexPattern,
+	}
+	functionToTest := builtinfunctions.GetFunctions()["bindConstants"]
+
+	for testName, testValue := range testItems {
+		tvLocal := testValue
+		t.Run(testName, func(t *testing.T) {
+			output, err := functionToTest.Call([]any{items, tvLocal})
+			assert.NoError(t, err)
+			assert.Equals(t, output.([]any), expectedOutBindConstants(tvLocal, items))
+		})
+	}
+
+	t.Run("no items in input list", func(t *testing.T) {
+		output, err := functionToTest.Call([]any{[]any{}, repeatedInputsMap})
+		assert.NoError(t, err)
+		assert.Equals(t, output.([]any), []any{})
+	})
+}
+
+func defaultPropertySchema(t schema.Type) *schema.PropertySchema {
+	return schema.NewPropertySchema(t, nil, false, nil, nil, nil, nil, nil)
+}
+
+func joinStrs(s1, s2 string) string {
+	return strings.Join([]string{s1, s2}, builtinfunctions.CombinedObjIDDelimiter)
+}
+
+// TestHandleTypeSchemaCombine tests the error cases for invalid input, and
+// the valid input cases create the expected behavior in the returned error
+// and returned type and type name, respectively.
```

Comment:
The second part of this sentence was hard to read. I think it's missing "that". Like "test that X happened".

Reply:
added it